src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerRequest.Managed.cs

```diff
@@ -383,32 +380,46 @@ public bool KeepAlive
 
         public Guid RequestTraceIdentifier => Guid.Empty;
 
-        public IAsyncResult BeginGetClientCertificate(AsyncCallback requestCallback, object state)
+        private IAsyncResult BeginGetClientCertificateCore(AsyncCallback requestCallback, object state)
         {
-            Task<X509Certificate2> getClientCertificate = new Task<X509Certificate2>(() => GetClientCertificate());
-            return TaskToApm.Begin(getClientCertificate, requestCallback, state);
+            var asyncResult = new GetClientCertificateAsyncResult(this, state, requestCallback);
+
+            // The certificate is already retrieved by the time this method is called.
+            ClientCertState = ListenerClientCertState.Completed;
+            asyncResult.InvokeCallback(GetClientCertificate());
```

Comment:
Also, do we have the ability to test this?  I was under the impression https in the managed implementation wasn't working, in which case, how do we know that the client certificate support (which is only relevant for https) actually works with this refactoring?